gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -399,9 +407,56 @@ message VehiclePosition {
   // including both seated and standing capacity, and current operating regulations allow.
   // It is possible that the value goes over 100 if there are currently more passengers than what the vehicle was designed for.
   // The precision of occupancy_percentage should be low enough that you can't track a single person boarding and alighting for privacy reasons.
+  // If multi_carriage_status is populated with per-carriage occupancy_percentage, 
+  // then this field should describe the entire vehicle with all carriages accepting passengers considered.
   // This field is still experimental, and subject to change. It may be formally adopted in the future.
   optional uint32 occupancy_percentage = 10;
 
+  // Carriage specific details, used for vehicles composed of several carriages
+  message CarriageStatus {
+
+    // User visible label that may be shown to the passenger to help identify
+    // the carriage. Example: "7712", "Car ABC-32", etc...
+    optional string label = 1;
+
+    // Occupancy status for this given carriage, in this vehicle
+    optional OccupancyStatus occupancy_status = 2 [default = NO_DATA_AVAILABLE];
+
+    // Occupancy percentage for this given carriage, in this vehicle.
+    // Follows the same rules as "VehiclePosition.occupancy_percentage"
+    // Use -1 in case data is not available for this given carriage (as protobuf defaults to 0 otherwise)
+    optional int32 occupancy_percentage = 3 [default = -1];
+
+    // Identifies the order of this carriage with respect to the other
+    // carriages in the vehicle's list of CarriageStatus.
+    // The first carriage in the direction of travel must have a value of 1.
+    // The second value corresponds to the second carriage in the direction
+    // of travel and must have a value of 2, and so forth.
+    // For example, the first carriage in the direction of travel has a value of 1.
+    // If the second carriage in the direction of travel has a value of 3,
+    // consumers will detect an issue with the sequencing.
+    // Even if protobuf keeps sequencing, it will help with feed validation
+    optional uint32 carriage_sequence = 4;
```

Comment:
How is switch-back / zig-zag / leversal / changing ends / turnround / turning / to horse over envisioned within a single trip?

Reply:
@skinkie My interpretation is that the sequence and label combinations should match whatever is shown to riders (i.e., the ground truth) for the given carriage configuration at the `VehiclePosition.timestamp`. So between two `VehiclePosition` updates, the sequence and label combinations could change, which indicates a change in configuration.

Reply:
@barbeau this is not sustainable. This means that ever time the train changes direction it either is enforced to become a new trip (which is not enforced in GTFS-static) or arbitrary need to reverse all the values in RT.

Reply:
Keep in mind a `label` is something that's customer facing.  If carriages have digital outward facing signage, the label may change at the time of train divide / amalgamate / switch-backs.

Could this be modelled by the `id` field?  You'll still need to reverse the carriage sequence at the time of the switch-back, but the consumer will be able to track flip through an idempotent reference to the physical carriage.  Consumers can also track carriages for divide and amalgamate operations.

It doesn't solve conveying these operations ahead of time, but at least it can be modelled and you'll know after it happens.

Reply:
@jxeeno Yes, if `id` persists across sequence changes, you should be able to detect these types of changes in real-time. So adding the `id` field seems like a reasonable approach to me.

Reply:
@barbeau @jxeeno I would even suggest to use the coach registration number here.

Reply:
We need to consider and handle the case -
For car ordering or car position, it will be done regardless of the direction of travel -
For many trains and train operators, generally, the train Consists always remains the same, and stay in the same order, with the car adjacent to the locomotive always being termed as Car # 1 and the next car as # 2, then Car # 3 and so on. 
So, the reporting of Car # 1 may need to remain the same, particularly with the AMTRAK Capitol Corridor CCJPA. In some other transit systems, it may not be the same.
Even though the train may be going in either direction, the locomotive is many times always on one side and would be either pulling or pushing (behind the cars) the train coaches.
Car 1 is always adjacent to the locomotive, regardless of the direction of travel.

Reply:
> We need to consider and handle the case -
> For car ordering or car position, it will be done regardless of the direction of travel -
> For many trains and train operators, generally, the train Consists always remains the same, and stay in the same order, with the car adjacent to the locomotive always being termed as Car # 1 and the next car as # 2, then Car # 3 and so on.
> So, the reporting of Car # 1 may need to remain the same, particularly with the AMTRAK Capitol Corridor CCJPA. In some other transit systems, it may not be the same.
> Even though the train may be going in either direction, the locomotive is many times always on one side and would be either pulling or pushing (behind the cars) the train coaches.
> Car 1 is always adjacent to the locomotive, regardless of the direction of travel.

For multi-car occupancy, “order of Car numbering may not be flipped every time the train direction of travel is changed”.
Carriage Sequence is always going to be 1, 2, 3, 4, 5
The first occurrence of the repeated field will represent the first car or vehicle, given the direction of travel is not always true generally with rail operations and passenger rail industry, specially in the case of AMTRAK Capitol Corridor and this needs to be accounted for, to have an operator view of the same.
There needs to be a special case for this consideration.

Reply:
@harshverma2000 I think you're referring to how the operator numbers carriages internally? The internal numbering doesn't need to match the "external" number following the rules defined here in GTFS-realtime - the operator would just need to keep a mapping between the two internally to ensure the right numbers are published to GTFS-realtime. There are certainly cases that the current GTFS-realtime model doesn't address, but IMHO it seems adequate for the purpose of showing occupancy per-carriage to the end user. @lfontolliet and @wafisher (and others) can certainly weigh in if they see any obstacles so far in implementation.

Reply:
Agreed with @barbeau. The operator would need to know the sequence of carriage for a given segment of a trip, but keeping a mapping between the internal identifier and their external facing sequence seems to be easier to handle in the operator's systems, compared to surfacing a complex logic in the standard.

Reply:
Thanks @barbeau and @lfontolliet for your thoughts and comments.  
The idea of keeping a mapping between the internal identifier and their external facing sequence to handle the operator's systems is excellent.
I believe operators could do that.  What could be important is for the purpose of understanding occupancy-by-carriage position by various end users - they would need to know the direction-of-travel in order to understand the mapping and this point has to be documented for operator-enduser understanding. Direction-of-travel field could help end users realize this mapping...